app/models/embedded_ansible_worker/runner.rb

```diff
@@ -60,13 +60,21 @@ def do_exit(*args)
     Thread.exit
   end
 
-  def update_embedded_ansible_manager
-    ansible = ManageIQ::Providers::EmbeddedAnsible::AutomationManager.first_or_initialize
-    server  = MiqServer.my_server(true)
-    ansible.default_endpoint.url = URI::HTTPS.build(:host => server.hostname, :path => "/ansibleapi/v1")
-    ansible.name = "Embedded Ansible"
-    ansible.zone = server.zone
-    ansible.save!
+  def update_embedded_ansible_provider
```

Comment:
Once this "way" is verified to be correct, my vote is to move this (in a followup PR) so we can do `ManageIQ::Providers::EmbeddedAnsible::Provider.pin_to_server(MiqServer.my_server(true)`

It feels to me that the EmbeddedAnsible::Provider should have this concern since it's truly the provider we need to pin to a server whenever the role is moved to a new server.